### DISCUSSION TITLE
Some Omissions & Bugs

### DISCUSSION THREAD

**Discussion by Cmaxster** [pwillworth/dfkreport#9] [General]
Hello, thanks again for making this incredible tool. I'm not sure if it's still in active development, but I noticed some possible bugs and thought it would be helpful to point out or get clarification here if they were intended.

1) I noticed that when minting a Jewel-LP Token, in the CSV I noticed both the amount of Jewel and ONE deposited in their respective columns, but the actual value of "Jewel-LP" is omitted.. (It's actually listed in the events column, but more like a comment and not accessible as raw data.. basically not accounted as a received token of value) 

I was under the impression that Jewel-LP is a token of monetary value just like questing items, etc. It's possible I might be wrong about how Jewel-LP works, so please correct me if it's irrelevant and intended in the current implementation.

2) I noticed that gas fees were omitted from calculations. I know gas fees generally amount to less than one cent, but they do add up to a small sum over a year, and probably should be reported. Was this an intentional omission?

3) There may be some formula parsing errors in the "gains" column of the generated "Summary" CSV. I noticed a couple "0E-28" "0E-29" instead of number values.

**Comment by pwillworth** [comment]
Hi again, glad the tool is helping you out!  Definitely still developing it, though most development lately has been keeping up with latest changes to DFK.
1. I chose not to track LP tokens as an asset, only the real crypto currencies they are place holding for.  This may be debatable, but I haven't seen any indication that they should be treated as a currency themselves.  They are more like a placeholder, not an asset.  However, the system does count the swap for the LP as a taxable event for those currencies you put in the pool, then tracks your gains as well when you remove it.  Another issue with LP tokens is their prices in fiat terms aren't really tracked, so I'm not sure how you would even put a value on them, it would be pretty convoluted.
2. Omitting gas fees was not intentional and I plan to add representation for them, but it just didn't make it into the initial feature set considering how negligible they are on Harmony.
3. Sometimes with all the calculations using floating decimals an almost infinitely small number can be left in some places and should be considered 0.  It is something that should be cleaned up though, will add it to the list.

**Reply by Cmaxster** [reply]
Awesome thanks so much for the detailed response. That's interesting about LP Tokens. I started looking into it myself and I wasn't even sure which tokens to treat as valued assets and which to treat as placeholder. Your assessment makes sense I think.

I see what you mean about gas fees and the calculations using floating decimals. It sounds like you've got it in progress so I'll look forward to seeing the next update :)
